nixos/modules/tasks/auto-upgrade.nix

```diff
@@ -109,7 +109,7 @@ in {
       '';
     }];
 
-    system.autoUpgrade.flags = [ "--no-build-output" ]
+    system.autoUpgrade.flags = optionals (cfg.flake == null) [ "--no-build-output" ]
       ++ (if cfg.flake == null then
         (if cfg.channel == null then
```

Comment:
Wouldn't it be nicer to put the flag in here instead?

`[ "--no-build-output" ] ++ (if cfg.channel == null then`

Reply:
Right.

Actually, I much more prefer to get rid of the if...then as well, and use optionals.